infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -119,21 +112,29 @@ private void openTables() {
     public Set<String> getStreamsToReplicate() {
         try {
             currentVersion = logReplicationConfigAdapter.getVersion();
-            Set<String> fetchedStreams = logReplicationConfigAdapter.fetchStreamsToReplicate();
+            Set<String> streams = logReplicationConfigAdapter.fetchStreamsToReplicate();
+            // Add registryTable to the streams
+            String registryTable = getFullyQualifiedTableName(
+                    CORFU_SYSTEM_NAMESPACE, TableRegistry.REGISTRY_TABLE_NAME);
+            streams.add(registryTable);
+
+            // Add protoBufDescriptorTable to the streams
+            String protoTable = getFullyQualifiedTableName(
+                    CORFU_SYSTEM_NAMESPACE, TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME);
+            streams.add(protoTable);
 
             return IRetry.build(IntervalRetry.class, () -> {
-                Set<String> streams = new HashSet<>();
                 try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
                     VersionResult result = verifyVersionResult(txn);
                     if (result.equals(VersionResult.UNSET) || result.equals(VersionResult.CHANGE)) {
                         boolean isUpgraded = result.equals(VersionResult.CHANGE);
                         // Case of upgrade or initial boot: clear table entries and sync with plugin info
-                        clearTables();
+                        log.info("Clearing plugin version table");
+                        pluginVersionTable.clearAll();
```

Comment:
A small question: why not just update the version table instead of clearing and then writing to it?

Reply:
Good point! I thought previously the clear operation was mainly for Name_Table. With only version table I agree we could just do an update.

Reply:
Sorry I forgot the point and just recalled it:
* For clearing the version table, we need to differentiate the case between version unset and version change. In the version change case we do a forced snapshot sync. For version unset (LR just bootstrapped) we just update version table.
* The txn here for the whole block is, we need to make the read and write operation to version table automatic, otherwise there could be race conditions between the nodes within the same cluster.

Reply:
The clear was mainly to get rid of any writes done outside LR(any accidental changes to the tables).  It is a remote possibility and we are only using the version key.  So it is fine to reset and overwrite it instead of clearing it completely.